client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx

```diff
@@ -227,27 +239,59 @@ const PersonalPurchase = ( {
 					{ translate( 'Continue with Jetpack Stats for free' ) }
 				</ButtonComponent>
 			) : (
-				<ButtonComponent
-					variant="primary"
-					primary={ isWPCOMSite ? true : undefined }
-					onClick={ () =>
-						gotoCheckoutPage( {
-							from,
-							type: 'pwyw',
-							siteSlug,
-							adminUrl,
-							redirectUri,
-							price: subscriptionValue / MIN_STEP_SPLITS,
-						} )
-					}
-				>
-					{ continueButtonText }
-				</ButtonComponent>
+				<StatsPurchaseActions
+					isWPCOMSite={ isWPCOMSite }
+					buttonText={ continueButtonText }
+					onCheckoutClick={ handleCheckoutRedirect }
+					onPostponeClick={ handleCheckoutPostponed }
+				/>
 			) }
 		</div>
 	);
 };
 
+function StatsPurchaseActions( props: any ) {
+	const allowPostponeAction = config.isEnabled( 'stats/checkout-flows-v2' );
+
+	return (
+		<div className={ `${ COMPONENT_CLASS_NAME }__actions` }>
+			<StatsPurchaseButton { ...props } />
+			{ allowPostponeAction && <StatsPurchasePostponeButton { ...props } /> }
+		</div>
+	);
+}
+
+function StatsPurchaseButton( props: any ) {
+	const ButtonComponent = props.isWPCOMSite ? CalypsoButton : Button;
+	const translate = useTranslate();
+	const newButtonText = translate( 'Contribute and continue' );
+	const buttonText = config.isEnabled( 'stats/checkout-flows-v2' )
+		? newButtonText
+		: props.buttonText;
+
+	return (
+		<ButtonComponent
+			variant="primary"
+			primary={ props.isWPCOMSite ? true : undefined }
+			onClick={ props.onCheckoutClick }
+		>
+			{ buttonText }
+		</ButtonComponent>
+	);
+}
+
+function StatsPurchasePostponeButton( props: any ) {
+	const ButtonComponent = props.isWPCOMSite ? CalypsoButton : Button;
```

Comment:
Almost all of the code in the proposed button components is duplicated, and it wouldn't have to be here if the buttons stayed in the main `PersonalPurchase` component. This increases bundle size and will be more significant if we continue to do it in many components.

Reply:
I'll look at moving the code back inline now that it's mostly working. I find it much easier to reason about page logic with smaller, more specific components but I recognize that isn't the popular style in the Stats code base.